spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/ChannelBindingAutoConfiguration.java

```diff
@@ -41,34 +33,20 @@
  *
  * @author Dave Syer
  * @author Marius Bogoevici
+ * @author Ilayaperumal Gopinathan
  */
 @Configuration
 @ConditionalOnBean(BindingService.class)
 @EnableConfigurationProperties(DefaultPollerProperties.class)
-@AutoConfigureBefore(EndpointAutoConfiguration.class)
+@Import(ChannelsEndpointConfiguration.class)
```

Comment:
We need `ChannelsEndpointConfiguration` (rename to `ChannelsEndpointAutoConfiguration`) to `spring.factories` rather than importing it here (among other things, explicit import renders `AutoConfigureBefore` moot, I think).